src/opentimelineio/stackAlgorithm.cpp

```diff
@@ -123,10 +125,49 @@ _flatten_next_item(
     }
 }
 
+// make all tracks the same length by adding a gap to the end if they are shorter
+static void
+_normalize_tracks_lengths(std::vector<Track*>& tracks,
+                          TrackRetainerVector& tracks_retainer,
+                          ErrorStatus*         error_status)
+{
+    RationalTime duration;
+    for (auto track: tracks)
+    {
+        duration = std::max(duration, track->duration(error_status));
+        if (is_error(error_status))
+        {
+            return;
+        }
+    }
+
+    for(int i = 0; i < tracks.size(); i++)
+    {
+        Track *track = tracks[i];
+        RationalTime track_duration = track->duration(error_status);
+        if (track_duration < duration)
+        {
+            track = static_cast<Track*>(track->clone(error_status));
```

Comment:
In reading this I got confused about the reuse of `track` as both the old one and the new one. Maybe `old_track` and `new_track` could make that more clear?